homeassistant/components/bayesian/binary_sensor.py

```diff
@@ -290,16 +308,12 @@ def _record_entity_observations(self, entity):
         local_observations = OrderedDict({})
 
         for entity_obs in self.observations_by_entity[entity]:
-            platform = entity_obs["platform"]
+            platform: str = str(entity_obs.platform)
```

Comment:
I don't think you need to set the type hint here. `entity_obs.platform` is already a string.
```suggestion
            platform = entity_obs.platform
```